server/utils/gsea.py

```diff
@@ -58,16 +84,23 @@ def extract_symbols(x):
             stop_loop_time = time.time()
             execution_time = stop_loop_time - start_loop_time
             print(f"Execution time: {execution_time} seconds")
-            signature=pd.DataFrame(df)
-            
-            # run enrichment analysis
-            start_gsea_time = time.time()
-            if __name__ == "__main__":
-               result = blitz.gsea(signature, msigdb_library).T
-               print ("result:",result.to_json())
-            stop_gsea_time = time.time()   
-            gsea_time = stop_gsea_time - start_gsea_time
-            print (f"GSEA time: {gsea_time} seconds")
+            try: # Extract ES data to be plotted on client side
+               geneset_name=json_object['geneset_name'] # Checks if geneset_name is present, if yes it indicates the server request is for generating the image. It retrieves the result.pkl file and generates the image without having to recompute gsea again.
+               result = pd.read_pickle(os.path.join(cachedir,"result.pkl"))
```

Comment:
won't work. multiple requests will corrupt and break it

Reply:
Support for random file names for both the pickle file and the image file have been added.